### ISSUE TITLE
Individual log messages not going to next line

### ISSUE THREAD

**Issue by CS-5** [sirupsen/logrus#1021]
This isn't necessarily a bug and the issue might not be related to Logrus at all, but I figured I'd post this anyway to see if I could get some advice.

I am not doing anything out of the ordinary, simple `log.WithField(...).Info(...)` statements with the colored text formatter (using JSON formatting doesn't make a difference, same issue). The contents of the messages are correct and function as expected, however each message is not always on it's own line.

For example:

I am seeing this
```
INFO[0000] Loaded Datasource ABC			dsName=ABC dsPath=/d/e/f
INFO[0000] Loaded Datasource XYZ			dsName=XYZ dsAddr=127.0.0.1INFO[0000] Datasource ABC has no valid data.			dsName=ABCINFO[0000] Datasource XYZ has no valid data.			dsName=XYZ
```

When I expect to see this:
```
INFO[0000] Loaded Datasource ABC			dsName=ABC dsPath=/d/e/f
INFO[0000] Loaded Datasource XYZ			dsName=XYZ dsAddr=127.0.0.1
INFO[0000] Datasource ABC has no valid data.			dsName=ABC
INFO[0000] Datasource XYZ has no valid data.			dsName=XYZ
```

I was thinking this might have some weird terminal line-wrapping thing causing this, but I've tried this in multiple terminal environments with and without line-wrapping enabled and the same issue exists. Is this a Logrus issue after all or is there something I'm overlooking?

**Comment by pradykaushik** [comment]
Can you provide the sample code that would help me reproduce this issue?

**Comment by CS-5** [comment]
Due to the nature of the project, I can't include much of the code, however I am loading and connecting to various data-sources in a loop when my program initializes. As part of this loop, the below logs information similar to what I've mentioned above:

```go
		log.WithFields(logrus.Fields{
			"dsName":    ds.Meta.Name,
			"dsType":    ds.Meta.Type,
			"dsPath":    ds.Meta.Path,
			"dsAddress": ds.Meta.Addr,
		}).Infof("Loaded datasource: %s\n", ds.Meta.Name)
```

`ds` is a struct pretty much consisting of only strings. What is unusual is that the first few log lines appear correctly but the following lines do not. There does not seem to be a specific reason why (the contents of the fields or order of the messages do not make a difference).

My Logrus initialization code is the following:
```go
	logrus.SetOutput(os.Stdout)
	log = logrus.New()

	logrus.SetLevel(logrus.DebugLevel)
	log.SetFormatter(&logrus.TextFormatter{
		ForceColors: true,
	})

	if !env.Debug {
		logrus.SetLevel(logrus.InfoLevel)
		log.SetFormatter(&logrus.JSONFormatter{})

		...
	}
```

**Comment by freeformz** [comment]
Is there any chance you can construct a simplified reproducer ?
Please also provide the version (number or commit sha) of logrus that you are experiencing this with.

**Comment by CS-5** [comment]
The first half of my week was packed and I created this issue in hopes of someone discovering a similar issue. After spending some time testing today, I determined this was the result of the hot-reloading script I was using which captured the `stdout` of my application and replayed it (loosing some of the message formatting in the process.

With all that being said, consider this issue closed and resolved with no problems related directly to Logrus.
